### PR TITLE
setup-environment: add layer search paths argument

### PR DESCRIPTION
**Pull request by mdurnev**
From https://github.com/MentorEmbedded/adit-custom/pull/580

### CODE REVIEW

File: scripts/setup-mel-builddir [hunk 1]
```diff
@@ -41,6 +41,8 @@ Options:
   -b BUILDDIR  Specify the build directory to create (default: 'build')
   -o MELDIR    Specify the root of the MEL install, where you can find bitbake
                and the layers
+  -a PATHS     Specify colon separated list of additional directories, where
+               you can find layers
```
Comment (kergoth) [inline scripts/setup-mel-builddir]: This should indicate that wildcards are valid, in some fashion, and might also want to mention what directories are searched/checked by default -- i.e. "what is this in addition *to*?"

File: scripts/setup-mel-builddir [hunk 2]
```diff
@@ -124,7 +126,7 @@ configure_for_machine () {
     echo 'BBLAYERS = "\' >> $BUILDDIR/conf/bblayers.conf
     layersfile=$(mktemp setup-mel-builddir.XXXXXX)
 
-    layerpaths="*:*/*:meta*/*/*:$MELDIR/*:$MELDIR/*/*:$MELDIR/meta*/*/*"
+    layerpaths="*:*/*:meta*/*/*:$MELDIR/*:$MELDIR/*/*:$MELDIR/meta*/*/*${DIRS:+:}$DIRS"
```
Comment (kergoth) [inline scripts/setup-mel-builddir]: `DIRS` is 1) too vague a name, 2) inconsistent with the case of our existing commandline argument variables, and 3) isn't set to a default value along with our other commandline option variables, meaning its value would also be picked up from the user's environment if set, which is particularly risky given the generic variable name. Please name it something more appropriate and set a default value, thanks!

**Comment by mdurnev** [comment]
@kergoth, thank you very much for your comments. The PR has been updated. Is it better now?

**Comment by Noor-Ahsan** [comment]
@kergoth are you still planning to use -a option for this. have you considered using different option as we discussed on IRC

**Comment by kergoth** [comment]
Having thought about it, I agree that -a isn't the most intuitive choice here, I think -L as you suggested (-L for search path, -l for what to search for, like the linker) or -p (for path, rather than -a for additional) are better options. I'll adjust it before I merge, so @mdurnev doesn't need to mess with this further. Thanks for the reminder.

**Comment by mdurnev** [comment]
@kergoth, I'd like https://github.com/MentorEmbedded/adit-custom/pull/580 to be in sync with this PR. I want to use the same option in adit-custom. Just let me know which option to use, and I can modify both PRs.

**Comment by kergoth** [comment]
Understood. Let's go with -p. As it supports and uses wildcards, the semantics differ from that of -L, which could be confusing.

**Comment by mdurnev** [comment]
@kergoth, the PR has been updated.

**Pull request merged**